nsdb-cluster/src/test/scala/io/radicalbit/nsdb/cluster/coordinator/ReadCoordinatorGlobalAggregatedStatementsSpec.scala

```diff
@@ -19,14 +19,82 @@ package io.radicalbit.nsdb.cluster.coordinator
 import io.radicalbit.nsdb.cluster.coordinator.mockedData.MockedData._
 import io.radicalbit.nsdb.common.protocol._
 import io.radicalbit.nsdb.common.statement._
-import io.radicalbit.nsdb.common.{NSDbDoubleType, NSDbLongType}
+import io.radicalbit.nsdb.common.{NSDbDoubleType, NSDbLongType, NSDbStringType}
 import io.radicalbit.nsdb.protocol.MessageProtocol.Commands._
 import io.radicalbit.nsdb.protocol.MessageProtocol.Events._
 
 class ReadCoordinatorGlobalAggregatedStatementsSpec extends AbstractReadCoordinatorSpec {
 
   "ReadCoordinator" when {
 
+    "receive a select containing a global aggregation but empty Result set" should {
+      "execute it successfully with a count" in {
+        probe.send(
+          readCoordinatorActor,
+          ExecuteStatement(
+            SelectSQLStatement(
+              db = db,
+              namespace = namespace,
+              metric = LongMetric.name,
+              distinct = false,
+              fields = ListFields(
+                List(Field("*", Some(CountAggregation)))
+              ),
+              Some(Condition(EqualityExpression("surname", AbsoluteComparisonValue(NSDbStringType("yorke")))))
+            )
+          )
+        )
+        val expected = awaitAssert {
+          probe.expectMsgType[SelectStatementExecuted]
+        }
+        expected.values shouldBe Seq(Bit(0, 0L, Map.empty, Map("count(*)" -> NSDbLongType(0L))))
+      }
+
+      "execute it successfully with an average" in {
+        probe.send(
+          readCoordinatorActor,
+          ExecuteStatement(
+            SelectSQLStatement(
+              db = db,
+              namespace = namespace,
+              metric = LongMetric.name,
+              distinct = false,
+              fields = ListFields(
+                List(Field("*", Some(AvgAggregation)))
+              ),
+              Some(Condition(EqualityExpression("surname", AbsoluteComparisonValue(NSDbStringType("yorke")))))
+            )
+          )
+        )
+        val expected = awaitAssert {
+          probe.expectMsgType[SelectStatementExecuted]
+        }
+        expected.values shouldBe Seq(Bit(0, 0L, Map.empty, Map("avg(*)" -> NSDbLongType(0L))))
+      }
+
+      "execute it successfully with a min()" in {
+        probe.send(
+          readCoordinatorActor,
+          ExecuteStatement(
+            SelectSQLStatement(
+              db = db,
+              namespace = namespace,
+              metric = LongMetric.name,
+              distinct = false,
+              fields = ListFields(
+                List(Field("*", Some(MinAggregation)))
+              ),
+              Some(Condition(EqualityExpression("surname", AbsoluteComparisonValue(NSDbStringType("yorke")))))
+            )
+          )
+        )
+        val expected = awaitAssert {
+          probe.expectMsgType[SelectStatementExecuted]
+        }
+        expected.values shouldBe Seq(Bit(0, 0L, Map.empty, Map("min(*)" -> NSDbLongType(Long.MaxValue))))
```

Comment:
I guess that we need to return an empty bit here

Reply:
Tackled it with current commit